rusoto/core/src/signature.rs

```diff
@@ -391,24 +400,34 @@ fn to_hexdigest<T: AsRef<[u8]>>(t: T) -> String {
     h.as_ref().to_hex().to_string()
 }
 
+fn build_scheme(region: Region) -> String {
+    match region {
+        Region::Local(_) => "http".to_owned(),
+        _                => "https".to_owned()
+    }
+}
+
 fn build_hostname(service: &str, region: Region) -> String {
     //iam has only 1 endpoint, other services have region-based endpoints
     match service {
         "iam" => {
             match region {
+                Region::Local(hostname) => hostname.to_owned(),
                 Region::CnNorth1 => format!("{}.{}.amazonaws.com.cn", service, region),
                 _ => format!("{}.amazonaws.com", service),
             }
         }
         "s3" => {
             match region {
+                Region::Local(hostname) => hostname.to_owned(),
                 Region::UsEast1 => "s3.amazonaws.com".to_string(),
                 Region::CnNorth1 => format!("s3.{}.amazonaws.com.cn", region),
                 _ => format!("s3-{}.amazonaws.com", region),
             }
         }
         _ => {
             match region {
+                Region::Local(hostname) => hostname.to_owned(),
```

Comment:
Instead of duplicating this, consider matching over `(service, region)`, so your branches would be like:

```rust
(_, Region::Local(hostname)) => hostname.to_owned(),
("iam", Region::UsEast1) => format!("{}.{}.amazonaws.com.cn", service, region),
("iam", _) => format!("{}.amazonaws.com", service),
...
```